src/data/finset/basic.lean

```diff
@@ -1447,12 +1453,60 @@ lemma union_sdiff_self (s t : finset α) : (s ∪ t) \ t = s \ t := sup_sdiff_ri
 lemma sdiff_singleton_eq_erase (a : α) (s : finset α) : s \ singleton a = erase s a :=
 by { ext, rw [mem_erase, mem_sdiff, mem_singleton], tauto }
 
-@[simp] lemma sdiff_singleton_not_mem_eq_self (s : finset α) {a : α} (ha : a ∉ s) : s \ {a} = s :=
-by simp only [sdiff_singleton_eq_erase, ha, erase_eq_of_not_mem, not_false_iff]
+lemma erase_eq (s : finset α) (a : α) : s.erase a = s \ {a} := (sdiff_singleton_eq_erase _ _).symm
+
+lemma disjoint_erase_comm : disjoint (s.erase a) t ↔ disjoint s (t.erase a) :=
+by simp_rw [erase_eq, disjoint_sdiff_comm]
+
+lemma disjoint_of_erase_left (ha : a ∉ t) (hst : disjoint (s.erase a) t) : disjoint s t :=
+by { rw [←erase_insert ha, ←disjoint_erase_comm, disjoint_insert_right],
+  exact ⟨not_mem_erase _ _, hst⟩ }
+
+lemma disjoint_of_erase_right (ha : a ∉ s) (hst : disjoint s (t.erase a)) : disjoint s t :=
+by { rw [←erase_insert ha, disjoint_erase_comm, disjoint_insert_left],
+  exact ⟨not_mem_erase _ _, hst⟩ }
+
+@[simp] lemma inter_erase (a : α) (s t : finset α) : s ∩ t.erase a = (s ∩ t).erase a :=
+by simp only [erase_eq, inter_sdiff]
+
+@[simp] lemma erase_inter (a : α) (s t : finset α) : s.erase a ∩ t = (s ∩ t).erase a :=
+by simpa only [inter_comm t] using inter_erase a t s
```

Comment:
What do you think of `inter_erase_left` and `inter_erase_right`? I'd expect `erase_inter` to be `erase (inter _ _) _`

Reply:
What if I tell you this matches `set`/`finset` `.` `union_insert`/`insert_union`?

Reply:
That would be pretty convincing!